numpy/core/src/multiarray/compiled_base.c

```diff
@@ -795,7 +795,52 @@ static int sequence_to_arrays(PyObject *seq,
     return 0;
 }
 
-/* Inner loop for unravel_index */
+/*
+ * Force-cast an array into intp if it is empty. Fail otherwise.
+ *
+ * op        - Pointer to the array pointer. Will be modified if
+ *             recasting is needed.
+ * paramname - Name to be used for error string.
+ */
+static int _cast_to_intp_or_fail(PyArrayObject **op,
+                                 char *paramname)
+{
+    PyArray_Descr *dtype = PyArray_DescrFromType(NPY_INTP);
+    if (dtype == NULL) {
```

Comment:
I did it because it also was done in `arr_unravel_index()`. I have zero clue of the likelyhood of any particular error.